rfq/custom_channel_data.go

```diff
@@ -0,0 +1,46 @@
+package rfq
+
+// TODO(guggero): De-duplicate by putting into same shared package as the other
```

Comment:
Ah, I now remember why I added this TODO... These types are also duplicated in the PoC. The types will reside in `tapchannelmsg` as well, so perhaps we should indeed keep the TODO for now.